### PR TITLE
gh-122273: Support PyREPL history on Windows

### PR DESCRIPTION
**Pull request by vstinner**
* Issue: gh-122273

### CODE REVIEW

File: Lib/site.py [hunk 1]
```diff
@@ -498,40 +498,55 @@ def register_readline():
...
 
     import atexit
+
+    try:
+        try:
+            import readline
+        except ImportError:
+            readline = None
+        else:
+            import rlcompleter  # noqa: F401
```
Comment (ambv) [inline Lib/site.py:508]: Can this import ever fail?

**Review by ambv — APPROVED** [review]
Tested on Windows 11 with PowerShell and cmd.exe, works as intended.

File: Lib/site.py (hunk 1 above)
Comment (vstinner) [reply Lib/site.py:508]: I don't know. In case of doubt, I prefer to keep the old behavior: catch and ignore ImportError.

**Comment by vstinner** [comment]
Merged, thanks for the review @ambv.

**Pull request merged**